examples/placeholder.cpp

```diff
@@ -0,0 +1,14 @@
+#include <rmx/rmx.hpp>
+#include <iostream>
+
+int main(int argc, const char** argv)
+{
+    rmx::Mutex<int> value(42);
+    auto guard = value.lock();
```

Comment:
Maybe add a way to unlock the mutex without dropping it? Would that need to poison the state of the guard object?

(would probably want to add an "is-locked" assertion to the value accessors? is that possible?)